lib/system.nim

```diff
@@ -2792,14 +2792,14 @@ else:
         yield T(res)
         inc(res)
     else:
-      var res: T = T(a)
+      var res = a
       while res <= b:
         yield res
         inc(res)
 
   iterator `..<`*[S, T](a: S, b: T): T {.inline.} =
     mixin inc
-    var i = T(a)
```

Comment:
But this one is correct, ``S`` can be unequal to ``T`` but we need to start with ``T``.

Reply:
ouch, reverted